### ISSUE TITLE
`prefer-continue` and `no-break-in-nested-loop`: contradicting reports in nested `for` loops

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3705]
Given the following code snippet:

```js
for (const group of groups) {
  for (const item of group) {
    if (item.isValid) {
      validate(item);
      save(item);
    }
  }
}
```

`prefer-continue` wants to rewrite the nested loop to:

```js
if (!item.isValid) {
  continue;
}

validate(item);
save(item);
```

But then `no-break-in-nested-loop` doesn't like the `continue` statement in the nested loop.

[Playground](https://eslint-online-playground.netlify.app/#eNptkc9uwyAMxl8FcUkrNWkv06T0uD3ALttl7MASJ2IlgIBUnaq8+wymXfcnByB8/uHP9pkH323hJCenofkIvOWD9WzVWRMiG72dHbMDHcKanYUR8SZARZiuepFFVANbJaVR4UVq1V8FEY/pX0bI+np/uQ7y+PNqSRsuC99wCFqZ2GDGQY3kUU3O+sjOrIdBGXjIElvY4O3ERCG2RAiObxaAhCc9j8o8G9VZb4ipSKldluqZtCq5MXDKLKaSs877NeXqNfks1REb2lQty195pv0vL8UsG2L9rIFIakhV0K3zMICvsZSozAxVi1a9t74q4E2osfW7B3mosQADIUJfa2vdH4RypvVtvcf+Otkd5Jimbw32NrsXPH46ELzF02R7dCc4Akno4fgIDkwPplMQUkwp+NJ4wjSOOeA5Y9/irxZT7P1ds2t2gqfQPPXlCwvZ7EQ=)

There wouldn't be anything wrong in this chain of reports if the both rules weren't in the recommended preset.

I can see a few ways to approach this:
- Do not suggest rewriting with `continue` in cases like this. Not a total fan of this because I personally find using the `continue` a better option that the original.
- Allow `continue` to be used in nested loops, but still not `break` (add an option that would disallow only `break`, only `continue` or both for `no-break-in-nested-loop`).
- (could be combined with the previous suggestion): exclude `no-break-in-nested-loop` from recommended. The suggested rewrite doesn't seem very common to me, so fully satisfying the rule if it's adopted for the first time may result in non-trivial refactoring effort.
- ??? anything else.
